pkg/scheduler/factory.go

```diff
@@ -473,40 +466,17 @@ func MakeDefaultErrorFunc(client clientset.Interface, podQueue internalqueue.Sch
 			klog.Errorf("Error scheduling %v/%v: %v; retrying", pod.Namespace, pod.Name, err)
 		}
 
-		podSchedulingCycle := podQueue.SchedulingCycle()
-		// Retry asynchronously.
-		// Note that this is extremely rudimentary and we need a more real error handling path.
-		go func() {
-			defer utilruntime.HandleCrash()
-			podID := types.NamespacedName{
-				Namespace: pod.Namespace,
-				Name:      pod.Name,
-			}
-
-			// Get the pod again; it may have changed/been scheduled already.
-			getBackoff := initialGetBackoff
-			for {
-				pod, err := client.CoreV1().Pods(podID.Namespace).Get(context.TODO(), podID.Name, metav1.GetOptions{})
-				if err == nil {
-					if len(pod.Spec.NodeName) == 0 {
-						podInfo.Pod = pod
-						if err := podQueue.AddUnschedulableIfNotPresent(podInfo, podSchedulingCycle); err != nil {
-							klog.Error(err)
-						}
-					}
-					break
-				}
-				if apierrors.IsNotFound(err) {
-					klog.Warningf("A pod %v no longer exists", podID)
-					return
-				}
-				klog.Errorf("Error getting pod %v for retry: %v; retrying...", podID, err)
-				if getBackoff = getBackoff * 2; getBackoff > maximalGetBackoff {
-					getBackoff = maximalGetBackoff
-				}
-				time.Sleep(getBackoff)
-			}
-		}()
+		// Check if the Pod exists in informer cache.
+		cachedPod, err := podLister.Pods(pod.Namespace).Get(pod.Name)
+		if err != nil {
+			klog.Warningf("Pod %v/%v doesn't exist in informer cache: %v", pod.Namespace, pod.Name, err)
+			return
+		}
+		// As <cachedPod> is from SharedInformer, we need to do a DeepCopy() here.
+		podInfo.Pod = cachedPod.DeepCopy()
+		if err := podQueue.AddUnschedulableIfNotPresent(podInfo, podQueue.SchedulingCycle()); err != nil {
+			klog.Error(err)
```

Comment:
TODO: error or warning this?

Reply:
This should only happen if there was an incoming Update... I would say Warning is fine.